aten/src/ATen/functorch/BatchRulesViews.cpp

```diff
@@ -556,12 +556,28 @@ Tensor trace_decomp(const Tensor& tensor) {
   return tensor.diagonal().sum();
 }
 
+std::tuple<Tensor,optional<int64_t>> tril_batch_rule(
+    const Tensor& self,
+    int64_t diagonal = 0) {
+  TORCH_CHECK(self.dim() >= 2, "tril: The input tensor must have at least 2 dimensions.");
+  auto result = at::tril(self, diagonal);
+  return std::make_tuple(result, 0);
```

Comment:
```suggestion
  return std::make_tuple(std::move(result), 0);
```

Reply:
This optimisation is almost always negligible in the grand scheme of things. If we really want to enforce it all across we should add it to the linter.

Reply:
@lezcano Adding it to a linter is bit more difficult that it sounds, a lot of false positives / negatives. Further complicated by template forwarding etc.

Reply:
Then I think it's not worth it to reviewing PRs making this point. As discussed, all these optimisations are not useful and not worth the CI run unless they are in a very hot path.

Reply:
@lezcano there is work on a clang-tidy check that I've been experimenting with, but it's not quite ready yet: https://reviews.llvm.org/D137205